index.html

```diff
@@ -854,23 +862,64 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
 '!PERFORMANCE-TIMELINE-2/#dfn-performance-entry-buffer'>performance
 entry buffer</a>.</li>
 <li>Increase <a>resource timing buffer current size</a> by 1.</li>
+<li>Return.</li>
 </ol>
 </li>
-<li>If <a>resource timing buffer current size</a> is greater than
-or equal to <a>resource timing buffer size limit</a> and the
-<a>resource timing buffer full flag</a> is false, run the following
+<li>If <a>resource timing buffer full flag</a> is false, run the following
 substeps:
-<ol style='list-style-type: lower-latin;' data-link-for=
-"Performance">
-<li>Set the <a>resource timing buffer full flag</a> to true.</li>
+<ol style="list-style-type: lower-latin;">
+<li>Set <a>resource timing buffer full flag</a> to true.</li>
+<li>Queue a task to run <a>fire a buffer full event</a>.</li>
+</ol>
+<li>Add <i>new entry</i> to the
+<a>secondary entry buffer</a> and increment <a>resource timing secondary buffer
+current size</a> by 1.</li>
+</ol>
+
+<p>To <dfn>fire a buffer full event</dfn>, run the following steps:</p>
+<ol>
 <li><a data-cite="!DOM/#concept-event-fire">Fire an event</a>
 named <code>resourcetimingbufferfull</code> at the <a data-cite=
-"!HR-TIME-2/#idl-def-performance">Performance</a> object, with its
+"!hr-time-2/#idl-def-performance">Performance</a> object, with its
 <code>bubbles</code> attribute initialized to true.</li>
+<li>While <a>secondary entry buffer</a> is not empty, run the following substeps:
+<ol>
+<li>Let <i>entry</i> be the oldest <a>PerformanceResourceTiming</a> in
+<a>secondary entry buffer</a>.</li>
+<li>Add <i>entry</i> to the end of <a data-cite='!PERFORMANCE-TIMELINE-2/#dfn-performance-entry-buffer'>performance
+entry buffer</a>.</li>
+<li>Increment <a>resource timing buffer current size</a> by 1.</li>
+<li>Remove <i>entry</i> from <a>secondary entry buffer</a>.</li>
+<li>Decrement <a>resource timing secondary buffer current size</a> by 1.</li>
 </ol>
+<li>While <a>resource timing buffer current size</a> is larger than 
+<a>resource timing buffer size limit</a>, run the following substeps:
+<ol>
+<li>Let <i>excess entries before</i> be the <a>resource timing buffer current size</a> 
+minus <a>resource timing buffer size limit</a>.</li>
+<li><a data-cite="!DOM/#concept-event-fire">Fire an event</a>
+named <code>resourcetimingbufferfull</code> at the <a data-cite=
+"!HR-TIME-2/#idl-def-performance">Performance</a> object, with its
+<code>bubbles</code> attribute initialized to true.</li>
+<li>Let <i>excess entries after</i> be the <a>resource timing buffer current size</a> 
+minus <a>resource timing buffer size limit</a>.</li>
+<li>If <i>excess entries before</i> equals <i>excess entries after</i>, then while <a>resource timing buffer current size</a> is larger than <a>resource timing buffer size limit</a>, run the following substeps:
```

Comment:
This isn't quite right. There is no guarantee that the secondary buffer is empty at this point.
We need to start dropping things from the secondary buffer first.

Reply:
In step 2 of [fire the buffer full event](https://pr-preview.s3.amazonaws.com/yoavweiss/resource-timing/pull/168.html#dfn-fire-a-buffer-full-event) we clear out the secondary buffer. Then if the number of entries exceeds the buffer size, we fire the event synchronously, and if the buffer size remained the same, we drop entries.

How could the secondary buffer be non-empty at this point? Are you suggesting that "add a PerformanceResourceTiming entry" call happened while these sync steps are running?

Reply:
As I mentioned during F2F, firing of `resourcetimingbufferfull` event itself can result in more resource timing entries to be added; e.g. via a sync XHR.

Reply:
@rniwa - Fixed the processing to fire the copying from the secondary buffer after the sync event firing (and deleting everything that exceeds if the buffer is equal or larger).
PTAL